Exiled.Events/Patches/Events/Player/Left.cs

```diff
@@ -73,10 +73,34 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
                 new(OpCodes.Call, Method(typeof(Handlers.Player), nameof(Handlers.Player.OnLeft))),
             });
 
+            newInstructions.InsertRange(newInstructions.Count - 1, new CodeInstruction[]
+            {
+                new(OpCodes.Ldloc_S, player.LocalIndex),
+                new(OpCodes.Call, Method(typeof(Left), nameof(Left.ForceCleanPlayerInfo))),
+            });
+
             for (int z = 0; z < newInstructions.Count; z++)
                 yield return newInstructions[z];
 
             ListPool<CodeInstruction>.Shared.Return(newInstructions);
         }
+
+        private static void ForceCleanPlayerInfo(Player player)
+        {
+            if (Player.InstantiatedRolesToPlayers.TryGetValue(player.Role, out HashSet<Player> previousRolePlayers))
+            {
+                previousRolePlayers.Remove(player);
+            }
+
+            if (Player.InstantiatedTeamToPlayers.TryGetValue(player.Role.Team, out HashSet<Player> previousTeamPlayers))
+            {
+                previousTeamPlayers.Remove(player);
+            }
+
+            if (Player.InstantiatedSideToPlayers.TryGetValue(player.Role.Side, out HashSet<Player> previousSidePlayers))
```

Comment:
For here i would say to use Player.InstantiatedToPlayers[player.Role.Side].Remove(player)

Reply:
Not guaranteed to be there; hence; previous comment explains why there is a protection here

Reply:
So protection will be not needed ^^

Reply:
Done, thanks!

Reply:
Not done everywhere

Reply:
Done